src/co_tools/co_utils.py

```diff
@@ -0,0 +1,76 @@
+import os
+import subprocess
+from multiprocessing import cpu_count
+
+# local imports
+from .get_logger import LOGGER
+
+co_cpus = os.getenv("CO_CPUS")
+aws_batch_job_id = os.getenv("AWS_BATCH_JOB_ID")
+
+
+def get_cpu_limit(co_cpus=co_cpus, aws_batch_job_id=aws_batch_job_id):
+    LOGGER.debug(f"co_cpus: {co_cpus}\naws_batch_job_id: {aws_batch_job_id}")
+    if co_cpus:
+        return co_cpus
+    if aws_batch_job_id:
+        return 1
+    with open("/sys/fs/cgroup/cpu/cpu.cfs_quota_us") as fp:
+        cfs_quota_us = int(fp.read())
+    with open("/sys/fs/cgroup/cpu/cpu.cfs_period_us") as fp:
+        cfs_period_us = int(fp.read())
+    container_cpus = cfs_quota_us // cfs_period_us
+    # For physical machine, the `cfs_quota_us` could be '-1'
+    LOGGER.debug(f"container_cpus: {container_cpus}\ncpu_count(): {cpu_count()}")
+    return cpu_count() if container_cpus < 1 else container_cpus
+
+
+def get_dir_contents(dir: str = "../data"):
+    cmd = ["find", "-L", dir]
```

Comment:
Don't use subprocess here.

Reply:
Used glob instead. Turned out better since now we can log a warning message when no files or folders are in the folder you are searching in.

Reply:
I might be missing something, did you push your changes? It still is using find here.